File: csharp/剑指32从上到下打印二叉树II.cs

public partial class Solution
{

    public IList<IList<int>> LevelOrder1(TreeNode root)
    {
        Queue<TreeNode> nodes = new Queue<TreeNode>();
        List<IList<int>> res = new();
        nodes.Enqueue(root);

        while (nodes.Count>0)
        {
            int size = nodes.Count;
            List<int> temp = new();
            for (int i = 0; i<size;++i)
            {
                TreeNode crt = nodes.Dequeue();
                if(crt!=null)
                {
                    temp.Add(crt.val);
                    nodes.Enqueue(crt.left);
                    nodes.Enqueue(crt.right);
                }
            }
            if(temp.Count>0)
                res.Add(temp);
        }
        return res;
    }
}

File: python/main/剑指32从上到下打印二叉树II.py
from binarytree import TreeNode


def levelOrder(root: TreeNode) -> list:
    if(root==None):
        return []
    
    nodes=[root]
    res=[[root.val]]

    while(len(nodes)):
        crt=[]
        for node in nodes:
            if(node.left!=None):
                crt.append(node.left)
            if(node.right!=None):
                crt.append(node.right)
        nodes=crt

        if(len(crt)):
            res.append([i.val for i in crt])
    
    return res